python_modules/libraries/dagster-fivetran/dagster_fivetran/resources.py

```diff
@@ -888,7 +890,24 @@ def fetch_fivetran_workspace_data(
             schema_configs_by_connector_id=schema_configs_by_connector_id,
         )
 
+    def sync_and_poll(
+        self, context: Optional[Union[OpExecutionContext, AssetExecutionContext]] = None
+    ):
+        raise NotImplementedError()
+
+    def __eq__(self, other):
+        return (
+            isinstance(other, FivetranWorkspace)
+            and self.account_id == other.account_id
+            and self.api_key == other.api_key
+            and self.api_secret == other.api_secret
+        )
+
+    def __hash__(self):
+        return hash((self.account_id + self.api_key + self.api_secret))
+
 
+@lru_cache(maxsize=None)
```

Comment:
Caching `load_fivetran_asset_specs` with `functools.lru_cache` requires `FivetranWorkspace` to be hashable.

An alternative to caching `load_fivetran_asset_specs` would be to call it in a `FivetranWorkspace` cached property.

```python
class FivetranWorkspace(ConfigurableResource):
    ...
    @cache_property
    def asset_specs(
        self, 
        dagster_fivetran_translator: Type[DagsterFivetranTranslator] = DagsterFivetranTranslator
    ) -> Sequence[AssetSpec]:
        return load_fivetran_asset_specs(
            workspace=self, 
            dagster_fivetran_translator=dagster_fivetran_translator
        )
```

For the context, after [this](https://www.notion.so/dagster/Asset-integration-customization-options-11a18b92e462807c94c2f84675021120) and [this](https://www.notion.so/dagster/Power-BI-semantic-model-API-options-11918b92e4628006a25af3f58a6b48d9) discussions, we decided to detach spec loading from the resources, which is why I cached `load_fivetran_asset_specs` instead of caching a property.

Reply:
Tagging @benpankow @dpeng817 @schrockn @alangenfeld @OwenKephart for feedback on this specific comment and the PR description about caching the asset specs calls.

Reply:
i bias towards `@cached_method` on an object instance instead of `@lru_cache(maxsize=None)` on a top level function - I think its much easier to reason about when things are expected to be cached

Reply:
we went with cached_property for the `FivetranWorkspace` equivalent in Airlift - I think that's likely the right move here as well.